### PR TITLE
Cache depth texture based on usage

### PR DESCRIPTION
**Pull request by IceSentry**
# Objective

- Currently, the depth textures are cached based on the target. If multiple camera have the same target but a different `depth_texture_usage` bevy will just use the same texture and ignore that setting.

## Solution

- Add the usage as a cache key

### CODE REVIEW

**Review by JMS55 — APPROVED** [review]

**Review by mockersf — APPROVED** [review]

**Review by superdump — APPROVED** [review]

**Comment by superdump** [comment]
Looks like some cargo fmt is needed?

**Comment by IceSentry** [comment]
yeah, this is caused by rust 1.72.0 now formatting let-else. This PR should fix it https://github.com/bevyengine/bevy/pull/9562

**Comment by IceSentry** [comment]
@superdump I discussed about this with @Neo-Zhixing and they suggested to just OR all the usages per render target. This would mean we'll only create 1 texture per target like it's done currently. The only unknown is whether or not some of the usage flags could be incompatible with each other. I'm curious if you have any opinions on this.

**Comment by Neo-Zhixing** [comment]
Yeah so my recommendation is that we do it over two passes, the first pass collect the usage flags per render target, the second pass creates the textures.  That way you create the same number of textures with the flag reflecting what all of the cameras requested.

The problem we're trying to solve here is that if multiple cameras specify the same render target but with different flags, it'll just use the flag for the first camera and ignore the rest.

I don't think texture usages can conflict with one another. People are generally more concerned about "if this texture usage was supported by the texture format", and usage flags should always be compatible i think.

**Comment by IceSentry** [comment]
I've implemented the two pass suggestion in my latest commit

**Review by Neo-Zhixing — APPROVED** [review]

File: crates/bevy_core_pipeline/src/core_3d/mod.rs
Comment (JMS55) [inline crates/bevy_core_pipeline/src/core_3d/mod.rs:377]:
```suggestion
                let usage = *render_target_usage
```
I think the code is a bit cleaner to deref it earlier

Comment (IceSentry) [reply crates/bevy_core_pipeline/src/core_3d/mod.rs]: done

**Review by JMS55 — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
@superdump @mockersf this is well-reviewed, but I don't fully understand what's going on here (docs on `Camera3dDepthTextureUsage` would have been useful), so I'm going to defer to one of you to merge this.

**Comment by alice-i-cecile** [comment]
It's been a week, so I'm going to merge this on the basis of your approvals to unblock this fix.

**Pull request merged**
